core/src/blocks/data_source.rs

```diff
@@ -82,11 +82,21 @@ impl DataSource {
         filter: Option<SearchFilter>,
         target_document_tokens: Option<usize>,
     ) -> Result<Vec<Document>> {
+        // This check is meant to block access to "conversations" space through a
+        // datasource block in a dust app, which could lead to data leaks, see related issue
+        // https://github.com/dust-tt/tasks/issues/1658
+        // Only case in which this is allowed is for our packaged apps, via a system
+        // key, in particular "assistant-retrieval-v2" that needs access to the
+        // conversation space
```

Comment:
Can we just convey isSystem and not say anyting about conversations data source here :pray: